v0.5/tools/submission/submission-checker.py

```diff
@@ -0,0 +1,238 @@
+    for system_id in system_ids:
+        errors = []
+        fname = os.path.join("systems", system_id + ".json")
+        try:
+            with open(fname, "r") as f:
+                j = json.load(f)
+                # make sure all required sections/fields are in the meta data
+                for k, v in meta.items():
+                    z = j.get(k)
+                    if z is None:
+                        errors.append("{} has no section".format(fname, k))
+                    else:
+                        for field, fieldval in v.items():
+                            sz = j[k].get(field)
+                            if sz is None and fieldval == "required":
+                                errors.append("{} section {} field {} missing".format(fname, k, field))
+                # make sure no undefined sections/fields are in the meta data
+                for k, v in j.items():
+                    z = meta.get(k)
+                    if z is None:
+                        errors.append("{} has unknwon section {}".format(fname, k))
+                    else:
+                        for field, fieldval in v.items():
+                            sz = meta[k].get(field)
+                            if sz is None:
+                                errors.append("{} section {} field {} unknown".format(fname, k, field))
+
+        except Exception as ex:
+            errors.append("{} unexpected error {}".format(fname, ex))
+        if errors:
+            for i in errors:
+                log.error(i)
+        else:
+            log.info("{} OK".format(fname))
+        return errors
+
+
+def check_measurement_dir(good_submissions):
+    errors = []
+    for submission in good_submissions:
+        fname = os.path.join("measurements", submission)
+        if not os.path.exists(fname):
+            errors.append("{} directory missing".format(fname))
+        cols = submission.replace("\\", "/").split("/")
+        system_id = cols[0]
+        system_file = None
+        dirs, files = path_to_dict(fname)
+        for i in REQUIRED_MEASURE_FILES:
+            if i not in files[fname]:
+                errors.append("{} is missing {}".format(fname, i))
+        for i in files[fname]:
+            if i.startswith(system_id) and i.endswith(".json"):
+                system_file = i
+                break
+        if not system_file:
+            errors.append("{} is missing {}*.json".format(fname, system_id))
+        impl = system_file[len(system_id) + 1:-5]
+        code_dir = os.path.join("code", cols[1], impl)
+        if not os.path.exists(code_dir):
+            errors.append("{} is missing code dir {}".format(fname, code_dir))
+        log.info("{} OK".format(fname))
+
+    if errors:
+        for i in errors:
+            log.error(i)
+    return errors
+
+
+def main():
+    args = get_args()
+
+    script_path = os.path.dirname(sys.argv[0])
+    with open(os.path.join(script_path, "meta.json"), "r") as f:
```

Comment:
What is this `meta.json`? I can't find it in submission rules: https://github.com/mlperf/policies/blob/master/submission_rules.adoc

Reply:
Just a json template of the meta fields defined in the submission doc. 
Christine noted above ... my meta.json is a little behind the submission doc - will fix it today.

Reply:
That makes sense. Closing.